### PR TITLE
fix #6937 [ Opening a Browser from a message list that contains class side methods ] (with test)

### LINKED ISSUE #6937: Opening a Browser from a message list that contains class side methods
**Describe the bug**
Subscript out of bounds exception when browsing code

**To Reproduce**
1. implementors message list tat contains class methods (e.g. Go to the Playground, type "uniqueInstances", implementors of)
2. Click on one of the class side methods
3. Try to Browse (e.g. via the menu)

### PR DESCRIPTION
**Pull request by hogoww**
Thanks @Alisu for the work :)

### CODE REVIEW

**Comment by StephanEggermont** [comment]
Please rename. Describe the issue

**Comment by MarcusDenker** [comment]
Tried the CI artefact image: bug is fixed

**Comment by hogoww** [comment]
@MarcusDenker We had an infinite loop while trying out the Spec tests locally on Unix. 
Could we relaunch the CI, just in case?

File: src/Spec2-Adapters-Morphic/SpNotebookTabSelectorMorph.class.st [hunk 1]
```diff
@@ -53,7 +53,7 @@ SpNotebookTabSelectorMorph >> removeTabIndex: anInteger [
 		selectedIndex = anInteger
```
Comment (dionisiydk) [inline src/Spec2-Adapters-Morphic/SpNotebookTabSelectorMorph.class.st]:
```suggestion
		selectedIndex = anInteger & (selectedIndex <= self tabs size)
```

File: src/Spec2-Adapters-Morphic/SpNotebookTabSelectorMorph.class.st [hunk 2]
```diff
@@ -53,7 +53,7 @@ SpNotebookTabSelectorMorph >> removeTabIndex: anInteger [
...
 			ifTrue: [
 				self selectedIndex: 0. "to update new tab at current selected index"
-				self tabs ifNotEmpty: [self selectedIndex: anInteger ] ]
+				self tabs ifNotEmpty: [self selectedIndex: self selectedIndex - 1 \\ self tabs size + 1 ] ]
```
Comment (dionisiydk) [inline src/Spec2-Adapters-Morphic/SpNotebookTabSelectorMorph.class.st]:
```suggestion
				self tabs ifNotEmpty: [ self selectedIndex: selectedIndex ]]
```

**Review by dionisiydk — CHANGES_REQUESTED** [review]
I think it can be simpler. See code suggestion

**Comment by hogoww** [comment]
I haven't tried them out, but I trust you @dionisiydk =)

**Comment by hogoww** [comment]
errored test: #testErrorCaughtAndDefferedIfExceptionSignaledAtStartupWhenStartupUiManagerActive.
Does not look related to me.

**Review by dionisiydk — APPROVED** [review]
thanks

**Pull request merged**